File: makeBSTtreefromsortedarray.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
struct TreeNode* sortedArrayToBST(int* nums, int numsSize)
{
    struct TreeNode* createNode(int val) {
    struct TreeNode* newNode = (struct TreeNode*)malloc(sizeof(struct TreeNode));
    newNode->val = val;
    newNode->left = NULL;
    newNode->right = NULL;
    return newNode;
}
        struct TreeNode* sorted(int* num, int start, int end)
        {
            if (start>end)
            {
                return NULL;
            }
            int mid = (start+end)/2; 
            struct TreeNode* root = createNode(num[mid]);
            root->left= sorted(num, start, mid-1);
            root->right= sorted(num, mid+1, end);
            return root;
        }
    return sorted(nums, 0, numsSize-1); 
}

File: makeBSTtreefromsortedarray.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sortedArrayToBST(self, nums: List[int]) -> Optional[TreeNode]:
        def sort(nums, start, end):
            if start > end:
                return None
            mid = (start+end)//2
            root = TreeNode(nums[mid])
            root.left=sort(nums, start, mid-1)
            root.right=sort(nums, mid+1, end)
            return root
        return sort(nums, 0 , len(nums)-1)
